Common/FileUtil.cpp

```diff
@@ -0,0 +1,40 @@
+﻿/* Copyright (C) 2015 Rainmeter Project Developers
+ *
+ * This Source Code Form is subject to the terms of the GNU General Public
+ * License; either version 2 of the License, or (at your option) any later
+ * version. If a copy of the GPL was not distributed with this file, You can
+ * obtain one at <https://www.gnu.org/licenses/gpl-2.0.html>. */
+
+#include "StdAfx.h"
+#include "FileUtil.h"
+
+namespace FileUtil {
+
+std::unique_ptr<BYTE[]> ReadFullFile(const std::wstring& path, size_t* size)
+{
+	FILE* file = _wfopen(path.c_str(), L"rb");
+	if (!file)
+	{
+		return nullptr;
+	}
+
+	fseek(file, 0, SEEK_END);
+	*size = ftell(file);
```

Comment:
Since `size` is optional, it may be null. It might be easiest to do something like:
```cpp
size_t fileSize = ftell(file);
...
...
if (size)
{
	*size = fileSize;
}

return buffer;
```